src/components/VoteAdder.jsx

```diff
@@ -0,0 +1,55 @@
+import { useState } from "react";
+import { PatchVotesById } from "../utils/api";
+
+function VoteAdder({ article_id, votes }) {
+  const [currVotes, setCurrVotes] = useState(votes);
+  const [voted, setVoted] = useState(false);
+  const [error, setError] = useState(null);
+
+  function incVotes() {
+    if (!voted) {
+      setCurrVotes((currVotes) => currVotes + 1);
+      setVoted(true);
+
```

Comment:
Great functionality here! You have made it so the user can only vote once, excellent work 👍